Wikipedia/Code/WMFCVLInfo.h

```diff
@@ -0,0 +1,43 @@
+#import <Foundation/Foundation.h>
+
+@class WMFCVLColumn;
+@class WMFCVLSection;
+@class WMFCVLAttributes;
+@class WMFCVLInvalidationContext;
+@class WMFColumnarCollectionViewLayout;
+@class WMFCVLMetrics;
+@protocol WMFColumnarCollectionViewLayoutDelegate;
+
+/*!
+ @class        WMFCVLInfo
+ @abstract     A WMFCVLInfo is a snapshot of a WMFColumnarCollecitonViewLayout and the layout attributes of the items and supplementary views. It handles the layout and organization into colums based on the WMFCVLMetrics provided to it.
+ @discussion   ...
+ */
+@interface WMFCVLInfo : NSObject <NSCopying>
+
+@property (nonatomic, strong, nonnull, readonly) NSArray <WMFCVLColumn *> *columns;
+@property (nonatomic, strong, nonnull, readonly) NSArray <WMFCVLSection *> *sections;
+
+@property (nonatomic) CGSize boundsSize;
+@property (nonatomic) CGSize contentSize;
+
+/*!
+ Initialize a new WMFCVLInfo with the given metrics.
+ @param metrics     The metrics for the WMFCVLInfo
+ @return A WMFCVLInfo with the given metrics
+ */
+- (nonnull instancetype)initWithMetrics:(nonnull WMFCVLMetrics *)metrics NS_DESIGNATED_INITIALIZER;
+
+- (void)enumerateSectionsWithBlock:(nonnull void(^)(WMFCVLSection * _Nonnull section, NSUInteger idx, BOOL * _Nonnull stop))block;
```

Comment:
Is there an implied order for these enumerations? I assume its in order of index path, but not sure if that is actually true. 

Either way you may want to add a comment to these 2 methods to let the user know that it is in order or the order is undefined.

Related: 
Does idx == indexPath.section?